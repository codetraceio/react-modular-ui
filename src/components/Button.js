import React from 'react';
import AbstractComponent from './AbstractComponent';

export default class Button extends AbstractComponent {
  getModifierObject() {
    return {
      size: this.props.size,
      view: this.props.view,
      color: this.props.color,
      shape: this.props.shape,
      disabled: this.props.disabled,
      padding: this.props.padding,
      paddingLeft: this.props.paddingLeft,
      paddingRight: this.props.paddingRight
    };
  }

  render() {
    return (
      <div
        className={this.getBlockName('button', this.getModifierObject())}
        data-name={this.props.name}
        tabIndex="1"
        onClick={this.props.onClick}
      >
        {this.props.children}
      </div>
    )
  }
}
