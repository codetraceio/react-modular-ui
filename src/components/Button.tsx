import * as React from "react";

import { Modifiers, getBlockName } from "../services/componentService";

export interface ButtonProps {
  size?: string | number;
  view?: string;
  color?: string;
  shape?: string;
  disabled?: boolean;
  padding?: string | number;
  paddingLeft?: string | number;
  paddingRight?: string | number;
  name?: string;

  onClick?: (event: React.MouseEvent<HTMLDivElement>) => void;
}

export default class Button extends React.PureComponent<ButtonProps, {}> {
  getModifierObject(): Modifiers {
    return {
      size: this.props.size,
      view: this.props.view,
      color: this.props.color,
      shape: this.props.shape,
      disabled: this.props.disabled,
      padding: this.props.padding,
      paddingLeft: this.props.paddingLeft,
      paddingRight: this.props.paddingRight
    };
  }

  render() {
    return (
      <div
        className={getBlockName("button", this.getModifierObject())}
        data-name={this.props.name}
        tabIndex={1}
        onClick={this.props.onClick}
      >
        {this.props.children}
      </div>
    );
  }
}
