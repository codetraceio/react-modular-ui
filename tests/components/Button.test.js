import * as React from "react";
import * as renderer from "react-test-renderer";

import Button from "../../src/components/Button";

describe("Button", () => {
  it("should render Button", () => {
    const tree = renderer.create(
      <Button
        size="32"
        view="flat"
        color="primary"
        shape="rounded"
        disabled
      >Something</Button>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
