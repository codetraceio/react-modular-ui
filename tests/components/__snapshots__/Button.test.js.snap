// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render Button 1`] = `
<div
  className="c-button -size-32 -view-flat -color-primary -shape-rounded -disabled"
  tabIndex={1}
>
  Something
</div>
`;
